### PR TITLE
Split pr part three

### PR DESCRIPTION
**Pull request by michaelkernaghan**
split up the big PR

### CODE REVIEW

File: integration-tests/config.ts [hunk 1]
```diff
@@ -101,18 +101,18 @@
   signerConfig: {
     type: SignerType.EPHEMERAL_KEY as SignerType.EPHEMERAL_KEY,
     keyUrl: 'http://key-gen-1.i.tez.ie:3010/mondaynet',
-    requestHeaders: { Authorization: 'Bearer taquito-example' },
+    requestHeaders: { Authorization: 'Bearer taquito' },
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/config.ts:104]: its ok

[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/config.ts]: it was changed back to Bearer taquito-example

File: integration-tests/contract-batch-high-number-of-operations.spec.ts [hunk 2]
```diff
@@ -6,13 +6,13 @@ const crypto = require('crypto');
...
 
-    describe(`Test contract.batch containing a high number of operations using: ${rpc}`, () => {
+    describe(`Test contract.batch containing a high number of operations through contract api using: ${rpc}`, () => {
         beforeEach(async (done) => {
             await setup(true);
             done();
         });
 
-        it('Test batch with 150 operations', async (done) => {
+        it('Verify contract.batch with 150 operations', async (done) => {
```
Comment (zainen) [inline integration-tests/contract-batch-high-number-of-operations.spec.ts:15]: just want to check do you want `contract.batch`? or `contract batch`

**Review by zainen — APPROVED** [review]
LGTM nice test addition!

File: integration-tests/contract-big-map.spec.ts [hunk 3]
```diff
@@ -204,7 +209,7 @@ CONFIGS().forEach(({ lib, rpc, setup, knownBigMapContract }) => {
       done();
     });
 
-     it('Return undefined when BigMap key is not found', async (done) => {
+    it('Return undefined when BigMap key is not found', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/contract-big-map.spec.ts:212]: It seems like there are duplicated tests in this file and the `integration-tests/contract-big-map-access-by-index.spec.ts` file

File: integration-tests/contract-fetch-multiple-big-map-keys.spec.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    describe(`Test contract origination with multiple BigMap variations through contract api using: ${rpc}`, () => {
+        /**  The contract code doesn't have annotations, so the keys of the storage and of the bigmap are indexes. */
+
+        beforeEach(async (done) => {
+            await setup();
+            done();
+        });
+
+        test('Verify contract.originate for a contract with a BigMap with multiple values to be indexed (also fetching the Storage/BigMap)', 2, async (done: () => void) => {
```
Comment (roxaneletourneau) [inline integration-tests/contract-fetch-multiple-big-map-keys.spec.ts:18]: jest-retries and `done: () => void` can be removed

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]

File: integration-tests/contract-batch.spec.ts [hunk 5]
```diff
@@ -28,7 +29,7 @@ CONFIGS().forEach(({ lib, rpc, setup, knownBaker, knownContract, createAddress }
             done();
         });
 
-        test('Simple transfers with origination (where the code in Michelson format)', async (done) => {
+        test('Verify contract.batch simple transfers with origination code in Michelson format', async (done: () => void) => {
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-batch.spec.ts:32]: removed them

File: integration-tests/contract-batch.spec.ts [hunk 6]
```diff
@@ -5,12 +5,13 @@ import { MANAGER_LAMBDA, OpKind } from '@taquito/taquito';
...
     const Tezos = lib;
-    describe(`Test contract.batch using: ${rpc}`, () => {
+
+    describe(`Test contract.batch through contract api using: ${rpc}`, () => {
         beforeEach(async (done) => {
             await setup();
             done();
         });
-        test('Simple transfers with origination (where the code in JSON Michelson format)', async (done) => {
+        test('Verify contract.batch simple transfers with origination code in JSON Michelson format',  async (done: () => void) => {
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-batch.spec.ts:14]: removed

File: integration-tests/contract-batch.spec.ts [hunk 7]
```diff
@@ -68,7 +69,7 @@ CONFIGS().forEach(({ lib, rpc, setup, knownBaker, knownContract, createAddress }
             done();
         })
 
-        test('Simple transfers with bad origination', async (done) => {
+        test('Verify handling of contract.batch simple transfers with bad origination', async (done: () => void) => {
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-batch.spec.ts:72]: removed

File: integration-tests/contract-batch.spec.ts [hunk 8]
```diff
@@ -93,7 +94,7 @@ CONFIGS().forEach(({ lib, rpc, setup, knownBaker, knownContract, createAddress }
             done();
         });
 
-        test('Test batch from account with low balance', async (done) => {
+        test('Verify transfer and origination for contract.batch simple transfers from an account with low balance', async (done: () => void) => {
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-batch.spec.ts:97]: removed

File: integration-tests/contract-batch.spec.ts [hunk 9]
```diff
@@ -120,7 +121,7 @@ CONFIGS().forEach(({ lib, rpc, setup, knownBaker, knownContract, createAddress }
             done();
         });
 
-        test('Chain contract calls', async (done) => {
+        test('Verify contract.batch simple transfers with chained contract calls', async (done: () => void) => {
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-batch.spec.ts:124]: removed

File: integration-tests/contract-batch.spec.ts [hunk 10]
```diff
@@ -147,7 +148,7 @@ CONFIGS().forEach(({ lib, rpc, setup, knownBaker, knownContract, createAddress }
             done();
         });
 
-        test('Batch transfers and method call', async (done) => {
+        test('Verify contract.batch of simple transfers and a contract entrypoint call using the array notation with kind', async (done: () => void) => {
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-batch.spec.ts:151]: removed

File: integration-tests/contract-batch.spec.ts [hunk 11]
```diff
@@ -162,29 +163,5 @@ CONFIGS().forEach(({ lib, rpc, setup, knownBaker, knownContract, createAddress }
             expect(batchOp.status).toEqual('applied');
             done();
         });
-
-        test('Batch multiple originations and get contract address info from getOriginatedContractAddresses member function', async (done) => {
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-batch.spec.ts:166]: restored the test and changed the description to:

File: integration-tests/contract-big-map.spec.ts (hunk 3 above)
Comment (michaelkernaghan) [reply integration-tests/contract-big-map.spec.ts:212]: sorry can't find any duplicated tests in those two files

File: integration-tests/contract-fetch-multiple-big-map-keys.spec.ts (hunk 4 above)
Comment (michaelkernaghan) [reply integration-tests/contract-fetch-multiple-big-map-keys.spec.ts:18]: removed

File: integration-tests/contract-batch-register-global-constant.spec.ts [hunk 12]
```diff
@@ -11,9 +11,13 @@ CONFIGS().forEach(({ lib, rpc, setup }) => {
             done();
         });
 
+<<<<<<< HEAD
```
Comment (roxaneletourneau) [inline integration-tests/contract-batch-register-global-constant.spec.ts:14]: Merge conflict marker encountered.

Comment (michaelkernaghan) [reply integration-tests/contract-batch-register-global-constant.spec.ts:14]: resolved the merge conflict

**Pull request merged**